lib/phoenix/socket/transport.ex

```diff
@@ -365,33 +365,40 @@ defmodule Phoenix.Socket.Transport do
     end)
   end
 
-  defp parse_origin(origin) do
+  defp parse_origin("http://*." <> orig),  do: parse_origin("http://" <> orig, :wildcard)
```

Comment:
So I would change the implementation. Instead of detecting if it is a wildcard here, I would just change compare_host? down below. We don't need to detect the type upfront.